### DISCUSSION TITLE
Request path does not exists and Fastapi is calling a different route

### DISCUSSION THREAD

**Discussion by thomas-michels** [fastapi/fastapi#10762] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import APIRouter, Depends
from fastapi.responses import JSONResponse
from fastapi.encoders import jsonable_encoder
from app.core.settings.app import AppSettings
from app.core.config import get_app_settings
from app.api.services import OrganizationServices, UserServices
from app.api.composer import organization_composer, user_composer
from app.api.domain import Organization, OrganizationInDB, UserInDB
from app.api.shared_schemas import Response, ResponseWithData, OrganizationLogin, Logon
from app.api.exceptions import InvalidOrganization
from app.core.logging import logger

router = APIRouter(prefix="/v2/organizations", tags=["Organization"])

@router.put("", responses={200: {"model": OrganizationInDB}})
async def reactive_org(
    services: OrganizationServices = Depends(organization_composer),
    settings: AppSettings = Depends(get_app_settings),
):
    return "test"
```

### Description

I trying to call an inexistent route called by `/v2/organizations/settings`, but fastapi calls another route called `/v2/organizations`. Requests have the same method, both are `PUT`.

![image](https://github.com/tiangolo/fastapi/assets/57485730/30aa0128-30f1-4a86-ad15-58a1ffd07dca)

I think there is a bug on Fastapi, because that behavior is very strange. The right behavior is the return status code 404 and default response body `{"detail":"Not Found"}`.

### Operating System

Windows

### FastAPI Version

0.104.1

### Pydantic Version

1.10.6

### Python Version

Python 3.10.11

**Comment by n8sty** [comment]
Unless you provide a minimally reproducible example it's unlikely you'll get help here

**Comment by YuriiMotov** [comment]
Works well in current version:

```python
from fastapi import APIRouter, FastAPI
from fastapi.testclient import TestClient

router = APIRouter(prefix="/v2/organizations", tags=["Organization"])


@router.put("")
async def reactive_org():
    return "test"


app = FastAPI()

app.include_router(router=router)


def test_():
    client = TestClient(app)
    resp = client.put("/v2/organizations/settings")
    assert resp.status_code == 404

```

Actually, I couldn't reproduce this even with FastAPI 0.104.1. It was probably an issue in Starlette
